pkg/storage/internal/actions/process.go

```diff
@@ -215,24 +232,33 @@ func (p *process) newStatuses(args *wdk.ProcessActionArgs) (txStatus wdk.TxStatu
 	return
 }
 
-func (p *process) broadcastSingleTx(ctx context.Context, txID string) (*wdk.ProcessActionResult, error) {
-	sendStatus, err := p.getSendStatus(ctx, txID)
+func (p *process) broadcastTxs(ctx context.Context, txIDs []string) (*wdk.ProcessActionResult, error) {
+	if len(txIDs) == 0 {
+		return &wdk.ProcessActionResult{}, nil
+	}
+
+	sendStatusesLookup, err := p.getSendStatuses(ctx, txIDs...)
 	if err != nil {
 		return nil, err
 	}
 
-	if sendStatus != wdk.SendWithResultStatusSending {
+	sendStatuses := maps.Values(sendStatusesLookup)
+	if seq.ContainsAll(sendStatuses, wdk.SendWithResultStatusUnproven) {
```

Comment:
Are you sure this is what you want to use?
Shouldn't It be rather:
```suggestion
	if seq.Every(sendStatuses, is.EqualTo(wdk.SendWithResultStatusUnproven)) {
```

Otherwise we won't broadcast any transaction if at least one would have status "unproven"

Reply:
Good point. I misunderstood the `ContainsAll` function purpose.